core/src/main/java/com/amplitude/core/platform/plugins/AmplitudeDestination.kt

```diff
@@ -7,10 +7,11 @@ import com.amplitude.core.events.IdentifyEvent
 import com.amplitude.core.events.RevenueEvent
 import com.amplitude.core.platform.DestinationPlugin
 import com.amplitude.core.platform.EventPipeline
+import com.amplitude.core.platform.NetworkConnectivityChecker
 import com.amplitude.core.platform.intercept.IdentifyInterceptor
 import kotlinx.coroutines.launch
 
-class AmplitudeDestination : DestinationPlugin() {
+class AmplitudeDestination(private val networkConnectivityChecker: NetworkConnectivityChecker? = null) : DestinationPlugin() {
```

Comment:
Could you let me know why you don't like changing the interface here? `AmplitudeDestination` is an internal class so I don't see any cons with it. 

Some reasons of the current design:
- Cleanness. The checker is only needed in `AmplitudeDestination` plugin not `Amplitude` class so it makes more sense to me to pass what the plugin needs to it rather than putting something unnecessary in `Amplitude` class
- Scalability. Android and JVM platform implementations are different. What actually passed in here for Android platform is `AndroidNetworkConnectivityChecker` which implements the interface and uses Android context. If we want to have the same feature for JVM, just implement this interface.

Reply:
It seems causing more friction and not cleanness to me. I am wondering if we could just use a plugin and an extra variable in amplitude instance to decouple and not change much.

Reply:
I'm afraid that it cannot be implemented by another plugin to prevent the default amplitude destination plugin from scheduling a flush.

Reply:
I think I agree with @qingzhuozhen here.

1. Add internal variable `connected` to `android.Amplitude`
2. Create new Before plugin that toggles amplitude.connected
3. We add a new Before plugin to android.Amplitude buildInternal
4. Destination plugin checks amplitude.isConnected

That will keep all the offline detection logic in the Plugin and reduce extra interfaces. wdyt?

Reply:
Thank you @qingzhuozhen and @justin-fiedler. It's more elegant and better decoupled. I implemented network checker as a plugin in https://github.com/amplitude/Amplitude-Kotlin/pull/171/commits/bcd0721544028a5036da43f867ca5a8d096974cf

Reply:
But I make `core.Configuration.isNetworkConnected` `public` as it's used in other modules (`com.amplitude.android.plugins`)